historical-adherence-query/java/Main.java

```diff
@@ -0,0 +1,204 @@
+        HistoricalAdherenceEventListener historicalAdherenceEventListener = new HistoricalAdherenceEventListener(myId);
+
+        // Subscribe to the listener
+        notificationHandler.addSubscription(historicalAdherenceEventListener);
+
+        // Connect to the web socket
+        notificationHandler.connect(true);
+
+        // Making an API call too soon after subscribing can result in the system thinking there are no subscribers.
+        TimeUnit.SECONDS.sleep(5);
+
+        // Workforce Management Api Instance for the request
+        WorkforceManagementApi wfmApiInstance = new WorkforceManagementApi();
+
+        String managementUnitId = "33333333-3333-3333-3333-333333333333"; // String | The management unit ID of the management unit
+
+        // List of all users in the management unit
+        List<User> users = wfmApiInstance.getWorkforcemanagementManagementunitUsers(managementUnitId).getEntities();
+
+        // List of the users to query
+        List<String> userIds = new ArrayList<String>();
+        userIds.add(users.get(0).getId());
+        userIds.add(users.get(1).getId());
+        userIds.add(users.get(2).getId());
+
+        // Calendar for setting the start and end date of the query
+        Calendar calendar = Calendar.getInstance();
+        calendar.set(Calendar.YEAR, 2020);
+        calendar.set(Calendar.MONTH, Calendar.JULY);
+        calendar.set(Calendar.DAY_OF_MONTH, 1);
+        calendar.set(Calendar.HOUR, 5);
+
+        Date startDate = calendar.getTime(); //Start date for the Historical Adherence query (2020-07-01T05:00:00.000Z)
+
+        // End date, here we are querying one day
+        calendar.set(Calendar.DAY_OF_MONTH, 2);
+        Date endDate = calendar.getTime(); //End date for the Historical Adherence query (2020-07-02T05:00:00.000Z)
+        String timeZone = "America/Chicago";
+
+        // Construct the body of the request using the variables set above
+        WfmHistoricalAdherenceQuery body = new WfmHistoricalAdherenceQuery(); // WfmHistoricalAdherenceQuery | body
+        body.setStartDate(startDate);
+        body.setEndDate(endDate);
+        body.setTimeZone(timeZone);
+        body.setUserIds(userIds);
+
+        try
+        {
+            // Send the request
+            WfmHistoricalAdherenceResponse pendingResult = wfmApiInstance.postWorkforcemanagementManagementunitHistoricaladherencequery(managementUnitId, body);
+            System.out.println(pendingResult);
+
+            // Save the operation ID
+            String operationId = pendingResult.getId();
+
+            // Wait for the query to complete
+            waitUntil(() -> historicalAdherenceEventListener.isCompleted(operationId), 15, 1000);
+
+            // Call the listener with the operation ID of the, now completed, request.
+            WfmHistoricalAdherenceCalculationsCompleteTopicWfmHistoricalAdherenceCalculationsCompleteNotice completedResult = historicalAdherenceEventListener.getResultForId(operationId);
```

Comment:
@PrinceNebulon I was chatting with Chuck and he was telling me that some of these really long names are being driven off the public API.  We should chat tomorrow and see if we can add some linter rules to flag when names are really long.

We might want to consider a "wall of shame" on any human created names that do not pass the length rules :)

Reply:
Not just some, but all. Every operation ID and every model class name is provided to the generator. None of them are chosen by a human.

Reply:
@PrinceNebulon So the robots have truly taken over. :)  I  think we should probably still look at working with Public ApI team and at least generate a report for violations.  Some of these names are just downright scary :).  I will open a ticket.  It can be a low priority ticket and maybe something we can provide as a part of a CI Genesys Cloud public API linter.

https://inindca.atlassian.net/browse/DEVCTR-246